packages/slice-machine/src/modules/selectedSlice/reducer.ts

```diff
@@ -0,0 +1,273 @@
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(deleteSliceWidgetMockCreator): {
+      if (!prevState) return prevState;
+
+      const updatedConfig = SliceMockConfig.deleteFieldMockConfig(
+        action.payload.mockConfig,
+        action.payload.variationId,
+        action.payload.widgetArea,
+        action.payload.newKey
+      );
+
+      const newModel = { ...prevState.Model, mockConfig: updatedConfig };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(generateSliceScreenshotCreator): {
+      if (!prevState) return prevState;
+
+      const newModel = {
+        ...prevState.Model,
+        screenshotUrls: action.payload.screenshots,
+      };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(generateSliceCustomScreenshotCreator): {
+      if (!prevState) return prevState;
+      const { variationId, screenshot } = action.payload;
+      const { Model: prevModel } = prevState;
+
+      const screenshots = prevModel.variations.reduce((acc, variation) => {
+        if (variation.id === variationId) {
+          return {
+            ...acc,
+            [variationId]: screenshot,
+          };
+        }
+        if (prevModel.screenshotUrls?.[variation.id]) {
+          return {
+            ...acc,
+            [variation.id]: prevModel.screenshotUrls?.[variation.id],
+          };
+        }
+        return acc;
+      }, {});
+
+      const newModel = { ...prevModel, screenshotUrls: screenshots };
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+      };
+    }
+    case getType(saveSliceCreator): {
+      if (!prevState) return prevState;
+
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(action.payload.state),
+      };
+    }
+    case getType(pushSliceCreator): {
+      if (!prevState) return prevState;
+      const newModel = {
+        ...prevState.Model,
+        __status: LibStatus.Synced,
+        initialScreenshotUrls: prevState.Model.screenshotUrls,
+        remoteVariations: prevState.Model.variations,
+      };
+
+      return { ...prevState, Model: updateTouchedAndStatus(newModel) };
+    }
+    case getType(copyVariationSliceCreator): {
+      if (!prevState) return prevState;
+      const { key, name, copied } = action.payload;
+      const newVariation = Variation.copyValue(copied, key, name);
+      const newModel = {
+        ...prevState.Model,
+        variations: prevState.Model.variations.concat([newVariation]),
+      };
+      return {
+        ...prevState,
+        Model: updateTouchedAndStatus(newModel),
+        variation: newVariation,
+      };
+    }
+    default:
+      return prevState;
+  }
+};
+
+const updateTouchedAndStatus = (model: SliceState) => {
```

Comment:
We do have some tickets for the Custom type store to take remove the "initialModel" and use the custom type list.
I think we should do the same here in another ticket.

Reply:
We'll have a spike for all of those before implementing something :D